deploy/cleanup.sh

```diff
@@ -3,54 +3,76 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
-set -x
-
 # parameters
-envCode=${1:-"aoi"}
+ENV_CODE=${1:-"aoi"}
+PRE_PROVISIONED_BATCH_ACCOUNT_NAME=${2:-$PRE_PROVISIONED_BATCH_ACCOUNT_NAME}
 
 NO_DELETE_DATA_RESOURCE_GROUP=${NO_DELETE_DATA_RESOURCE_GROUP:-"false"}
-DATA_RESOURCE_GROUP="${envCode}-data-rg"
+DATA_RESOURCE_GROUP="${ENV_CODE}-data-rg"
 
 NO_DELETE_MONITORING_RESOURCE_GROUP=${NO_DELETE_MONITORING_RESOURCE_GROUP:-"false"}
-MONITORING_RESOURCE_GROUP="${envCode}-monitor-rg"
+MONITORING_RESOURCE_GROUP="${ENV_CODE}-monitor-rg"
 
 NO_DELETE_NETWORKING_RESOURCE_GROUP=${NO_DELETE_NETWORKING_RESOURCE_GROUP:-"false"}
-NETWORKING_RESOURCE_GROUP="${envCode}-network-rg"
+NETWORKING_RESOURCE_GROUP="${ENV_CODE}-network-rg"
 
 NO_DELETE_ORCHESTRATION_RESOURCE_GROUP=${NO_DELETE_ORCHESTRATION_RESOURCE_GROUP:-"false"}
-ORCHESTRATION_RESOURCE_GROUP="${envCode}-orc-rg"
+ORCHESTRATION_RESOURCE_GROUP="${ENV_CODE}-orc-rg"
 
 NO_DELETE_PIPELINE_RESOURCE_GROUP=${NO_DELETE_PIPELINE_RESOURCE_GROUP:-"false"}
-PIPELINE_RESOURCE_GROUP="${envCode}-pipeline-rg"
+PIPELINE_RESOURCE_GROUP="${ENV_CODE}-pipeline-rg"
 
 if [[ ${NO_DELETE_DATA_RESOURCE_GROUP} != "true" ]] && [[ ${NO_DELETE_DATA_RESOURCE_GROUP} == "false" ]]; then
     set -x
     az group delete --name ${DATA_RESOURCE_GROUP} --no-wait --yes
+    az deployment sub delete --name ${DATA_RESOURCE_GROUP} --no-wait
     set +x
 fi
 
 if [ "${NO_DELETE_MONITORING_RESOURCE_GROUP}" != "true" ] && [ "${NO_DELETE_MONITORING_RESOURCE_GROUP}" == "false" ]; then
     set -x
     az group delete --name ${MONITORING_RESOURCE_GROUP} --no-wait --yes
+    az deployment sub delete --name ${MONITORING_RESOURCE_GROUP} --no-wait
     set +x
 fi
 
 if [ "${NO_DELETE_NETWORKING_RESOURCE_GROUP}" != "true" ] && [ "${NO_DELETE_NETWORKING_RESOURCE_GROUP}" == "false" ]; then
     set -x
     az group delete --name ${NETWORKING_RESOURCE_GROUP} --no-wait --yes
+    az deployment sub delete --name ${NETWORKING_RESOURCE_GROUP} --no-wait
     set +x
 fi
 
 if [ "${NO_DELETE_ORCHESTRATION_RESOURCE_GROUP}" != "true" ] && [ "${NO_DELETE_ORCHESTRATION_RESOURCE_GROUP}" == "false" ]; then
     set -x
     az group delete --name ${ORCHESTRATION_RESOURCE_GROUP} --no-wait --yes
+    az deployment sub delete --name ${ORCHESTRATION_RESOURCE_GROUP} --no-wait
     set +x
 fi
 
 if [ "${NO_DELETE_PIPELINE_RESOURCE_GROUP}" != "true" ] && [ "${NO_DELETE_PIPELINE_RESOURCE_GROUP}" == "false" ]; then
     set -x
     az group delete --name ${PIPELINE_RESOURCE_GROUP} --no-wait --yes
+    az deployment sub delete --name ${PIPELINE_RESOURCE_GROUP} --no-wait
     set +x
 fi
 
-set +x
\ No newline at end of file
+BATCH_ACCOUNT_POOL_NAME=${BATCH_ACCOUNT_POOL_NAME:-"${ENV_CODE}-data-cpu-pool"}
+NO_BATCH_ACCOUNT_POOL_DELETE=${NO_BATCH_ACCOUNT_POOL_DELETE:-"false"}
+if [[ ! -z "$PRE_PROVISIONED_BATCH_ACCOUNT_NAME" ]] && [ "${NO_BATCH_ACCOUNT_POOL_DELETE}" != "true" ] && [ "${NO_BATCH_ACCOUNT_POOL_DELETE}" == "false" ]; then
```

Comment:
If we are cleaning up the job and pool created, then we should also remove the RBACs added as part of the Batch Account switch to fully rollback all the changes done.

Reply:
I rechecked the batch-account and the access for managed identities which we were granting access on batch account got cleaned on their own as the managed identities are deleted after the test finishes.
Let me know if we are talking about some other managed identity.